linters/badtime/main.go

```diff
@@ -98,46 +107,66 @@ func handleImportPaths(importPaths []string, buildTags []string, callback callba
 
 	for _, pkg := range prog.InitialPackages() {
 		for _, file := range pkg.Files {
-			ast.Walk(newMapVisitor(fs, pkg.Types, callback), file)
+			ast.Walk(mapVisitor{
+				fs:               fs,
+				types:            pkg.Types,
+				mapCallback:      mapCallback,
+				equalityCallback: equalityCallback,
+			}, file)
 		}
 	}
 }
 
-func newMapVisitor(fs *token.FileSet, types map[ast.Expr]types.TypeAndValue, callback callbackFunc) mapVisitor {
-	return mapVisitor{
-		fs:    fs,
-		types: types,
-		callback: func(position token.Position, keyStr, valStr string) {
-			callback(position, path.Base(keyStr), path.Base(valStr))
-		},
-	}
-}
-
 type mapVisitor struct {
-	fs       *token.FileSet
-	types    map[ast.Expr]types.TypeAndValue
-	callback callbackFunc
+	fs               *token.FileSet
+	types            map[ast.Expr]types.TypeAndValue
+	mapCallback      mapCallbackFunc
+	equalityCallback equalityCallbackFunc
 }
 
 func (v mapVisitor) Visit(node ast.Node) ast.Visitor {
+	// Detect time.Time == time.Time
```

Comment:
do you want to split this into two separate visitors? weird that you're using `mapVisitor` to check binary expressions.

Reply:
I think its a little cleaner (and less wasteful) as one visitor. I'll just rename it